pandas/tests/dtypes/test_inference.py

```diff
@@ -834,6 +834,30 @@ def test_infer_dtype_datetime(self):
         arr = np.array([np.nan, "2011-01-01", pd.Timestamp("2011-01-02")])
         assert lib.infer_dtype(arr, skipna=True) == "mixed"
 
+    def test_infer_dtypes_datetimetz(self):
+
+        # starts with nan
```

Comment:
Can you split this into 3 different tests?

And for these `# starts with nan` test, could you use `pytest.mark.parameterize`